### PR TITLE
native: integrate activity

### PR DESCRIPTION
**Pull request by latter-bolden**
This covers the initial integration of `%activity` into the native app which means:

- moving our unreads infrastructure to pull from there vs the channels/chat agents
- issuing channel/thread/comment reads to `%activity` more consistently (we trigger reads as soon as you see the divider, but it sticks around for a delay or until you navigate away)
- implements volume controls for top level device notifications + granular muting at the DM/group & thread level
- adds in firehose subscription for DMs so we'll get live updates for posts and reactions now

This now also includes the work for the main activity feed (middle tab)

Fixes TLON-1749
Fixes TLON-1505
Fixes TLON-2084
Fixes TLON-2083

### CODE REVIEW

File: packages/ui/src/components/SwipableChatListItem.tsx
Comment (patosullivan) [inline packages/ui/src/components/SwipableChatListItem.tsx:26]: Nice

File: packages/ui/src/contexts/channel.tsx
Comment (patosullivan) [inline packages/ui/src/contexts/channel.tsx:23]:
```suggestion
      'Must call `useChannelContext` within an `ChannelProvider` component.'
```

**Review by patosullivan — APPROVED** [review]
Overall, looks great. Just a couple little things (mostly questions), but could be merged as-is afaict.

File: apps/tlon-mobile/src/navigation/NavBarView.tsx
Comment (latter-bolden) [reply apps/tlon-mobile/src/navigation/NavBarView.tsx]: Yeah, leaving undotted for now. Added a comment about it being intentional

File: packages/shared/src/db/keyValue.ts
Comment (latter-bolden) [reply packages/shared/src/db/keyValue.ts:1]: Good catch, I meant to add a `.web` version of that file. Figure we can get away with leaning on `localStorage` for that case?

File: packages/shared/src/api/postsApi.ts
Comment (dnbrwstr) [inline packages/shared/src/api/postsApi.ts:665]: My memory was that ub.Posts was generic + could handle posts or writs -- is that not the case?

File: packages/shared/src/db/queries.test.ts
Comment (dnbrwstr) [inline packages/shared/src/db/queries.test.ts:1]: This would be a nice test to keep if not super-painful.

File: packages/shared/src/db/schema.ts
Comment (dnbrwstr) [inline packages/shared/src/db/schema.ts:183]: ?

File: packages/ui/src/components/UnreadDot.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/UnreadDot.tsx:12]: Should be theme color -- usually even grayscale color change from dark to light me.

File: packages/ui/src/components/SwipableChatListItem.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/SwipableChatListItem.tsx:166]: Theme colors

File: packages/ui/src/components/SwipableChatListItem.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/SwipableChatListItem.tsx:69]: ?

File: packages/ui/src/components/SwipableChatListItem.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/SwipableChatListItem.tsx:36]: Use haptic utils?

File: packages/ui/src/components/ProfileScreenView.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/ProfileScreenView.tsx:111]: ?

File: packages/ui/src/components/NotebookPost/NotebookPost.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/NotebookPost/NotebookPost.tsx:62]: Can we pad in the wrapper instead of hardcoding?

File: packages/ui/src/components/ListItem.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/ListItem.tsx:311]: ?

File: packages/ui/src/components/ContentRenderer.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/ContentRenderer.tsx:807]: What type error does this fix? Seems like the added type would already be a valid post?

File: packages/ui/src/components/ContentRenderer.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/ContentRenderer.tsx:666]: ?

File: packages/ui/src/components/ContentRenderer.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/ContentRenderer.tsx:434]: Not necessary for this PR, but putting it out there: I wonder whether using styled context/variants would be a cleaner way to handle the conditional tweaks.

File: packages/ui/src/components/Activity/ActivityHeader.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/Activity/ActivityHeader.tsx:1]: Can this use standard screenheader for top bit?

File: packages/shared/src/urbit/channel.ts
Comment (dnbrwstr) [inline packages/shared/src/urbit/channel.ts:377]: ?

File: packages/shared/src/store/useInitialSync.ts
Comment (dnbrwstr) [inline packages/shared/src/store/useInitialSync.ts:4]: ? (commented)

File: packages/shared/src/store/dbHooks.ts
Comment (dnbrwstr) [inline packages/shared/src/store/dbHooks.ts:101]: Could these hooks be combined?

File: packages/ui/src/components/Activity/ActivityScreenView.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/Activity/ActivityScreenView.tsx:129]: Would recommend memoizing this as well as event handlers in `ActivityEventRaw`.

File: packages/ui/src/components/Activity/ActivitySourceContent.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/Activity/ActivitySourceContent.tsx:37]: Should memoize this stuff.

File: packages/ui/src/components/Activity/ChannelActivitySummary.tsx
Comment (dnbrwstr) [inline packages/ui/src/components/Activity/ChannelActivitySummary.tsx:122]: Where are you pulling design from? The one I see in Ochre uses standard sizes.

**Review by dnbrwstr — CHANGES_REQUESTED** [review]
Looking good! Let's discuss when you're around; some things we can probably skip, some we could divide and conquer.

File: apps/tlon-mobile/src/screens/ActivityScreen.tsx
Comment (patosullivan) [inline apps/tlon-mobile/src/screens/ActivityScreen.tsx:59]: Could we add this at the root of the app instead?

Comment (latter-bolden) [reply apps/tlon-mobile/src/screens/ActivityScreen.tsx:59]: Let's do both for now. I can update this to use the same component you have once that PR merges

File: packages/shared/src/db/keyValue.web.ts
Comment (latter-bolden) [reply packages/shared/src/db/keyValue.web.ts:1]: Was added as a stub after it was noted in review that `keyValue` will require a web adapter.

I don't care either way, could see leaving it for now or removing.

File: packages/ui/src/components/SwipableChatListItem.tsx
Comment (latter-bolden) [reply packages/ui/src/components/SwipableChatListItem.tsx:69]: I swear it was a cute feature flag once upon a time...

File: packages/ui/src/components/ContentRenderer.tsx
Comment (latter-bolden) [reply packages/ui/src/components/ContentRenderer.tsx:434]: Strong agree! 

Getting our display logic for chat messages, gallery blocks, and notebook posts to handle `viewMode` more comprehensively would be a cool followup project.

File: packages/ui/src/components/Activity/ChannelActivitySummary.tsx
Comment (latter-bolden) [reply packages/ui/src/components/Activity/ChannelActivitySummary.tsx:122]: see here:
https://www.figma.com/design/mUIcOGeELxWswuusYUnEiW/2024-Q2-Tlon.app?node-id=3026-48044&m=dev

these exact pixel values may not be correct, but our size steps definitely didn't work for rendering channel icons that tiny

File: apps/tlon-mobile/src/screens/ProfileScreen.tsx
Comment (latter-bolden) [reply apps/tlon-mobile/src/screens/ProfileScreen.tsx:19]: I can update it to pass around Expo constants, but I was trying to avoid `packages/ui` having to think about those.

**Review by dnbrwstr — APPROVED** [review]
Lgtm!

**Comment by latter-bolden** [comment]
Verified this continues to work as expected after the last batch of fixes + Dan's sync/volume work.

Out of an abundance of caution, I'm gonna hold off on merging anything else into `develop` until we get the `6.0.1` patch released, but....

this is ready to go  ✅

**Pull request merged**
